### PR TITLE
WIP Session UI

### PR DESCRIPTION
**Pull request by samleewy**
View commit `e458546` for the changes built upon @yungweezy Session UI!
Should be stable for now, haven't tested for checks yet.

### CODE REVIEW

File: src/main/java/seedu/address/ui/SessionListPanel.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        studentList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                System.out.println("Changes have been made to student");
+                sessionList.clear();
+                for (Student student : change.getList()) {
+                    sessionList.addAll(student.getListOfSessions());
+                }
+            }
+        });
```
Comment (JonahhGohh) [inline src/main/java/seedu/address/ui/SessionListPanel.java:43]: Do you mind explaining the listener part and what the change.next() does? I a bit catch no ball 😅 . 
Also, which is the part that calls the view to update the session list every time a session is added?

Comment (chwoozy) [reply src/main/java/seedu/address/ui/SessionListPanel.java:43]: The `change.next()` method controls the position of an iterator that goes through the discrete changes in the Student List. Whenever there is a change the listener catches these changes and triggers the contents of the `onChanged` function. The while loop will terminate after all tracked changes are iterated. Here's the [API](https://docs.oracle.com/javase/8/javafx/api/javafx/collections/ListChangeListener.Change.html)

Comment (samleewy) [reply src/main/java/seedu/address/ui/SessionListPanel.java:43]: I think can imagine the listener as something that's running in the "background" and always monitoring the `ObservableList<Student>`, so whenever there's any changes to the list of students, the code within the block will be invoked. As for the `change.next()`, if there are multiple changes made, they're represented as a 'separate change' (list of changes), so `change.next()` loops through it (might not be 100% correct in this part).

The `ListView<Session>` is actually bound to the `ObservableList<Session> sessionList`, so whenever there's a change in the sessionList, which is shown in this block of code you commented, the ListView will update as well.

_(might not be 100% correct)_

Comment (chwoozy) [reply src/main/java/seedu/address/ui/SessionListPanel.java:43]: Yeah, I believe the "calling" to update the ListView is abstracted away in the implementation of ObservableLists in JavaFx under the Observable collections.

I think the reason why we're using ObservableLists is properly summarised in this [article](https://www.odiestas.com/en/understanding-the-use-of-observablelist-on-javafx/).

A `ListView` bounded by `List` will not update upon changes, but a `ListView` bounded by `ObservableList` will.

**Comment by chwoozy** [comment]
Pulled to #67.

**Pull request closed without merging**